### PR TITLE
fix: Windows permission UX + adapter trust hardening

### PR DESCRIPTION
**Pull request by agent-refr**
> **AI-assisted change proposal.** Filed by agent driven by @Cervator via [GDD](https://siliconsaga.github.io/yggdrasil/gdd/).

## Summary

Windows/permission UX fixes and a trust-boundary hardening round, built across two workspaces with a cross-review pass.

**Permission UX (the #133 thread):**
- **Normalize quoted Windows path tokens before the backslash ask (#133).** The #130 escape hardening fail-closed to ask on any backslash before the allowlist, making the hook near-always-ask on a Windows workspace. Fully quoted drive-letter-rooted tokens (`"D:\Dev\..."`) are preserved as path data by Bash, so those — and only those — rewrite to forward slashes before Tier 1 and reach the allowlist normally. Bare backslash paths stay approval-required (Bash strips their backslashes, so a normalized match would diverge from what actually executes); escaped quotes, trailing/doubled backslashes, and non-path tokens still ask, and the redirect denies are checked first so a backslash can never downgrade a real redirect from deny to ask. The ask message now names the two recoveries (forward slashes, or quote the full path).
- **Session env-file carve-out for the #129 sub-agent prompt friction.** Sub-agents create their own `<name>.env` identity files at birth (the `ws commit --co-author-file` contract), and #129's sensitive-path ask fired on every one. A complete Write rides the scratch allow when: both requested and resolved parents are the sessions dir itself, the target is a single-segment `<name>.env`, the full content carries no guard-scope key (`GDD_K8S_*` stays a `ws k8s` ceremony), and it is the session's own `<sid>.env` or creates a file that does not exist yet. Partial Edits always ask (an Edit's `new_string` alone cannot prove the resulting file carries no guard key — it could retarget a guard *value*), as do overwrites of another session's existing file, guard keys, and non-env names.
- **`bash -n` syntax checks no longer trip the Kubernetes write floor** — `-n` anywhere in a short-flag cluster is parse-only, mirroring the existing `-c` inline exemption.
- **Windows suite portability:** `ws clone --add-to-ecosystem` pins adopted local source paths through `ws_native_path` instead of whatever MSYS env conversion produced, and a test's `rg` dependency became `grep -r` — first fully-green Windows suite runs.

**Trust-boundary hardening (scan findings):**
- Tighten command authorization around Git execution modifiers and Kubernetes command aliases so equivalent or path-prefixed spellings cannot inherit a broader read allowance: abbreviated Git long options (`--uplo=`, `--exec-p=` — including the `--exec-path` family) deny as prefixes of the dangerous set; k8s raw-read auto-allow is reserved for literal `kubectl`/`ws k8s` spellings; the Codex bridge normalizes quoted/escaped/`.exe` kubectl command words for classification while auditing the original.
- Require current realm approval before adapter-routed test or lint execution, and bind regular realm-owned helper files referenced by adapter commands into the trust fingerprint — a helper cannot change after approval without making trust stale.
- Reject adapter helper symlinks that leave the reviewed realm (including lexically-inside-but-canonically-outside paths through intermediate symlinks) while allowing unrelated documentation updates without renewed approval.
- Sanitize provider-supplied review text before terminal rendering, and validate all merged component keys against one name grammar before they become filesystem paths or trust-summary labels.
- Preserve GDD's intended local-operator trust model: these changes close avoidable authorization ambiguity without claiming isolation from a separate, already-approved hostile local process.

**Docs/notes:** CHANGELOG `[Unreleased]` caught up through #130/#131 plus this branch; GA readiness doc gains a 2026-07-16 sequencing note; hook README documents the Windows-path rule and the session-file carve-out.

## Review notes

- Realm approval remains semantic rather than revision-based. Adapter configuration and referenced executable helpers invalidate approval when changed; unrelated realm documentation does not.
- Kubernetes scope checks remain an accident-prevention boundary for a trusted local session. They do not attempt to pin manifest bytes against concurrent mutation by another approved local writer.
- Cross-workspace review pass on the hardening round found one escape — `--exec-path` was in the exact-spelling regex but not the abbreviation prefix set, so `git --exec-p=/tmp/evil log` slipped a broad `git:*` allow — fixed with a regression case in the abbreviation loop.

## Test plan

- [x] Full suite green on Windows 10 / Git Bash and on the second workspace (`ws test yggdrasil`; first fully-green Windows runs)
- [x] `ws audit-permissions` — clean (`[]`)
- [x] Backslash regression tests: quoted drive paths single- and double-quoted (allow), bare path (ask), `\"` (ask), trailing/doubled backslash (ask), redirect after a normalized path (deny), newline list with a Windows path (deny)
- [x] Session env carve-out tests: new sub-agent file (allow), own session file complete Write (allow), partial Edit of own file with a guard key present (ask), other session's existing file (ask), guard-scope keys in content (ask), non-env name (ask), backslash-spelled Write paths on both the allow and ask sides; the #129 baseline test unchanged
- [x] Live fresh-session smoke test on Windows: quoted backslash path, `bash -n`, and sub-agent identity-file creation all silent; the guard-key overwrite prompted as designed (audit-log verified — the log records the normalized `D:/` form)
- [x] Symlink trust rejections exercised where symlinks are real; capability-skipped on hosts where `ln -s` degrades to a copy
- [x] Guard: `bash -n` / `-nv` resolver exemption; full k8s guard + Codex bridge files green
- [x] Adapter trust: helper fingerprint change staleness, symlink and symlink-escape rejection, docs-only realm changes stay approved, `ws test`/`ws lint` refuse on stale trust
- [x] Git modifier abbreviations: prefix forms of every dangerous long option deny before broad allows (incl. `--exec-p`); unrelated complete long options still match normally
- [x] Shell syntax checks for every changed shell entry point and helper; `git diff --check` — clean

## Related

- Closes #133

## Summary by CodeRabbit

* **Security**
  * Realm approvals now fail closed when semantic configuration or referenced adapter files/tools change, with explicit rejection of realm-boundary escape attempts.
  * Hardened Kubernetes command and scope handling, including clearer denial behavior and safer redirect/wrapper recognition.
  * Improved protections for session env-file writes, Git execution modifiers, unsafe option parsing, and Windows backslash path ambiguities.
  * Sanitized provider-supplied terminal output to strip control characters.

* **Bug Fixes**
  * Improved Windows Git Bash compatibility for path handling (including `ws docker`) and `ws` adapter trust checks.

* **Documentation**
  * Expanded trust/safety and hook-path/session rules to match the updated behavior.

* **Tests**
  * Added/expanded Bats coverage for the above edge cases.

### CODE REVIEW

**Comment by agent-refr** [comment]
Migration note for existing workspaces: this PR expands the realm trust fingerprint to include adapter-referenced helper files, so a previously approved realm reads as stale after the first pull — ws commands that touch realm config fail closed until a one-time `ws realm use --trust <realm>` reapproval (verified live on this workspace while opening this PR). Same shape as the #129/#130 explicit-activation change already noted in the CHANGELOG.

File: .claude/hooks/gdd-permission-hook.sh [hunk 1]
```diff
@@ -805,6 +853,76 @@ if [[ "$tool_name" == "PowerShell" ]]; then
...
+                quote='"' core="${w:1:${#w}-2}"
+            fi
+            if [[ -n "$quote" ]] && _backslash_token_is_path "$core"; then
+                core="${core//\\//}"
+                w="${quote}${core}${quote}"
+                changed=1
+            fi
+        fi
+        rebuilt+="${rebuilt:+ }${w}"
+    done
```
[the remark this answers is not in the document]
Comment (agent-refr) [reply .claude/hooks/gdd-permission-hook.sh:912]: Deliberate scope, documented rather than fixed: the normalizer word-splits before inspecting tokens and will not re-implement shell quoting inside a security-sensitive classifier, so a quoted path containing spaces falls through to the ask tier (fail-closed, never fail-open). Spaced Windows paths are rare in this workspace, and the ask message plus the hook README now steer toward forward slashes, which never prompt. Revisit only if spaced paths become real recurring friction.

**Review by Cervator — APPROVED** [review]
